src/js/components/ColumnChooserMenu.js

```diff
@@ -36,66 +88,90 @@ export default class ColumnChooserMenu extends React.Component<Props> {
     return this.props.tableColumns.allVisible()
   }
 
-  showAllColumns = (e: Event) => {
+  deselectAllColumns = (e: Event) => {
+    e.stopPropagation()
+    this.props.dispatch(Columns.hideAllColumns(this.props.tableColumns))
+  }
+
+  selectAllColumns = (e: Event) => {
     e.stopPropagation()
     this.props.dispatch(Columns.showAllColumns(this.props.tableColumns))
   }
 
   onColumnClick(e: Event, column: TableColumn) {
     e.stopPropagation()
     if (column.isVisible) {
-      if (this.allVisible()) {
-        this.props.dispatch(Columns.hideAllColumns(this.props.tableColumns))
-        this.props.dispatch(Columns.showColumn(this.tableId(), column))
-      } else if (this.props.tableColumns.visibleCount() === 1) {
-        this.props.dispatch(Columns.showAllColumns(this.props.tableColumns))
-      } else {
-        this.props.dispatch(Columns.hideColumn(this.tableId(), column))
-      }
+      this.props.dispatch(Columns.hideColumn(this.tableId(), column))
     } else {
       this.props.dispatch(Columns.showColumn(this.tableId(), column))
     }
   }
 
-  className() {
-    return classNames("column-chooser-menu", {
-      "all-visible": this.props.tableColumns.allVisible()
-    })
-  }
-
   render() {
+    const {columnHeadersView} = this.props
     const columns = this.props.tableColumns.getColumns()
     const count = this.props.tableColumns.visibleCount()
-    const allVisible = this.props.tableColumns.allVisible()
+
+    const template = [
+      {
+        label: "On",
+        click: () => this.props.dispatch(Viewer.setColumnHeadersView("ON"))
+      },
+      {
+        label: "Off",
+        click: () => this.props.dispatch(Viewer.setColumnHeadersView("OFF"))
+      },
+      {
+        label: "Auto",
+        click: () => this.props.dispatch(Viewer.setColumnHeadersView("AUTO"))
+      }
+    ]
+
+    const openMenu = usePopupMenu(template)
+    const onClick = () => {
+      openMenu()
+    }
```

Comment:
I'd suggest using a simple `<select>` input for this. There is a styled version in the `common/forms/SelectInput.js` file.  It might look something like this:

```jsx
const view = useSelector(Viewer.getColumnHeadersView)
const onChange = (e) => dispatch(Viewer.setColumnHeadersView(e.target.value))

<SelectInput value={view} onChange={onChange}>
  <option value="ON">On</option>
  <option value="OFF">Off</option>
  <option value="AUTO">Auto</option>
</SelectInput>
```